engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -210,7 +227,18 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId, workflowDescript
   onTransition {
     case _ -> toState if toState.terminal =>
```

Comment:
Combine inside conditional statement with external if statement  if toState.Terminal &&  toState == WorkflowExecutionSuccessfulState, even when u need to case statements , that would not require if else ...